### ISSUE TITLE
Restore whitespaces in build directory name on jenkins

### ISSUE THREAD

**Issue by tmakar** [ElektraInitiative/libelektra#4923]
With this PR (https://github.com/ElektraInitiative/libelektra/pull/4895) we have introduced the go binding into our main repo. `Go` seems to have issue with spaces in any folder names etc. so it fails on builds / tests, etc. As we are using `build directory` as our build directory name and it has a space in the name it will lead to issues with `go`. We have therefore removed this whitespace now.

```groovy
return [(testName): {
    stage(testName) {
      withDockerEnv(image, dockerOpts) {
        // we use a space in the directory to test if paths are
        // properly escaped
        def buildDir='build directory'
```

With this issue the removal of whitespaces should be restored again. (we use a space in the directory to test if paths are properly escaped).

**Comment by markus2330** [comment]
Thanks for creating the issue.

What is the upstream bug? Otherwise we don't know when we can revert this change. The one I found https://github.com/golang/go/issues/1413 is already fixed?

**Comment by tmakar** [comment]
> Thanks for creating the issue.
> 
> What is the upstream bug? Otherwise we don't know when we can revert this change. The one I found [golang/go#1413](https://github.com/golang/go/issues/1413) is already fixed?

I took a look at https://github.com/golang/go/issues/1413. This one is about `GOROOT`. The one we need is https://github.com/golang/go/issues/16455, but it also seems to be "fixed". The solution seems to be to escape the name which contains spaces (generated `pkg-config` file) with `\`.

**Comment by markus2330** [comment]
Doesn't sound too hard, can you apply this fix for us?

**Comment by tmakar** [comment]
> Doesn't sound too hard, can you apply this fix for us?

Sure, will do :wink:

**Comment by markus2330** [comment]
fix depends on #4924
